services/app-api/handlers/banners/write.ts

```diff
@@ -16,23 +18,39 @@ export const writeBanner = handler(async (event, _context) => {
   } else if (!event?.pathParameters?.bannerId!) {
     throw new Error(NO_KEY_ERROR_MESSAGE);
   } else {
-    const body = JSON.parse(event!.body!);
-    const params = {
-      TableName: process.env.BANNER_TABLE_NAME!,
-      Item: {
-        key: event.pathParameters.bannerId,
-        createdAt: Date.now(),
-        lastAltered: Date.now(),
-        lastAlteredBy: event?.headers["cognito-identity-id"],
-        type: body.type,
-        title: body.title,
-        description: body.description,
-        link: body.link,
-        startDate: body.startDate,
-        endDate: body.endDate,
-      },
-    };
-    await dynamoDb.put(params);
-    return { status: StatusCodes.SUCCESS, body: params };
+    const unvalidatedPayload = JSON.parse(event!.body!);
+
+    const validationSchema = yup.object().shape({
+      key: yup.string().required(),
+      title: yup.string().required(),
+      description: yup.string().required(),
+      url: yup.string().url().notRequired(),
+      startDate: yup.number().required(),
+      endDate: yup.number().required(),
+    });
+
+    const validatedPayload = await validateData(
+      validationSchema,
+      unvalidatedPayload
+    );
+
+    if (validatedPayload) {
+      const params = {
+        TableName: process.env.BANNER_TABLE_NAME!,
+        Item: {
+          key: event.pathParameters.bannerId,
+          createdAt: Date.now(),
+          lastAltered: Date.now(),
+          lastAlteredBy: event?.headers["cognito-identity-id"],
+          title: unvalidatedPayload.title,
+          description: unvalidatedPayload.description,
+          link: unvalidatedPayload.link,
+          startDate: unvalidatedPayload.startDate,
+          endDate: unvalidatedPayload.endDate,
```

Comment:
Should these come from the `validatedPayload` object instead? Values should be the same and I think it reads better if we're storing the validated values

Reply:
Yes. To do.